package/kedro_viz/integrations/kedro/data_catalog_lite.py

```diff
@@ -0,0 +1,76 @@
+"""``DataCatalogLite`` is a custom implementation of Kedro's ``DataCatalog``
+to provide a MemoryDataset instance when running Kedro-Viz in lite mode.
+"""
+
+import copy
+from typing import Any, Optional
+
+from kedro.io.core import AbstractDataset, DatasetError, generate_timestamp
+from kedro.io.data_catalog import DataCatalog, _resolve_credentials
+from kedro.io.memory_dataset import MemoryDataset
+
+
+class DataCatalogLite(DataCatalog):
+    """``DataCatalogLite`` is a custom implementation of Kedro's ``DataCatalog``
+    to provide a MemoryDataset instance by overriding ``from_config`` of ``DataCatalog``
+    when running Kedro-Viz in lite mode.
+    """
+
+    @classmethod
+    def from_config(
+        cls,
+        catalog: Optional[dict[str, dict[str, Any]]],
+        credentials: Optional[dict[str, dict[str, Any]]] = None,
+        load_versions: Optional[dict[str, str]] = None,
+        save_version: Optional[str] = None,
+    ) -> DataCatalog:
+        datasets = {}
+        dataset_patterns = {}
+        catalog = copy.deepcopy(catalog) or {}
+        credentials = copy.deepcopy(credentials) or {}
+        save_version = save_version or generate_timestamp()
+        load_versions = copy.deepcopy(load_versions) or {}
+        user_default = {}
+
+        for ds_name, ds_config in catalog.items():
+            if not isinstance(ds_config, dict):
+                raise DatasetError(
```

Comment:
Could you pls explain this error handling more ? as i am a bit confused ... when it happens, how the user will understand?

Reply:
This is from the original DataCatalog implementation. This might happen when the config provided is not a dictionary, lets say 

```yaml
companies: "type: pandas.csvdataset"
```